### PR TITLE
add model match-pyramid

### PR DESCRIPTION
**Pull request by yinhaofeng**
add model match-pyramid and a readme profile

### CODE REVIEW

File: models/match/match-pyramid/config_train.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  device: cpu
+  save_checkpoint_interval: 1 # save model interval of epochs
+  save_inference_interval: 1 # save inference
+  save_checkpoint_path: "inference" # save checkpoint path
+  save_inference_path: "inference" # save inference path
+  save_inference_feed_varnames: [] # feed vars of save inference
+  save_inference_fetch_varnames: [] # fetch vars of save inference
+  init_model_path: "" # load model path
+  print_interval: 2
+  phases: phase_train
```
Comment (seiriosPlus) [inline models/match/match-pyramid/config_train.yaml:65]: `phases` is an instance of `list`?

File: models/match/match-pyramid/data/process.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    data = []
+    for line in open(filename):
+        line = line.strip().split()
+        data.append((int(line[0]), line[1], line[2]))
+    print('[%s]\n\tInstance size: %s' % (filename, len(data)))
+    return data
+
+
+Letor07Path = "."
+word_dict = read_word_dict(filename=Letor07Path + '/word_dict.txt')
```
Comment (seiriosPlus) [inline models/match/match-pyramid/data/process.py:68]: why declare a new var `Letor07Path ` ?

File: models/match/match-pyramid/model.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            lod_level=0)
+        return [sentence_left, sentence_right]
+
+    def embedding_layer(self, input):
+        """
+        embedding layer
+        """
+        if os.path.isfile(self.emb_path):
+            embedding_array = np.load(self.emb_path)
+            emb = fluid.embedding(
```
Comment (seiriosPlus) [inline models/match/match-pyramid/model.py:69]: use `fluid.layers.embedding`

File: models/match/match-pyramid/data/process.py (hunk 2 above)
Comment (seiriosPlus) [reply models/match/match-pyramid/data/process.py:68]: `os.path.join`

**Comment by yinhaofeng** [comment]
for merge

**Pull request closed without merging**